.github/workflows/prune-preview-deploys.yml

```diff
@@ -31,6 +31,12 @@ jobs:
       - name: Setup Fly IO
         uses: superfly/flyctl-actions/setup-flyctl@master
 
+      - name: Configure AWS credentials using OIDC
+        uses: aws-actions/configure-aws-credentials@v4
```

Comment:
you could test the permissions for this by briefly adding `  pull_request:` to the `on:` section at the top of this file